mantidproject/mantid
# Handle data collected beyond end of a frame

# Fixes 15825

LARMOR and SANS2D are starting to operate in two different time regimes for monitors and detectors. The data shows frame spillage, i.e. monitors further down the beam line measure data from the previous frame.

The problem is solved (for linearly spaced monitor workspaces) by duplicating the data set and allowing the user to specify the allowed wavelength range.

**To test:**
- Load the file LARMOR00009848_corrected.nxs which you can find here: /olympic/babylon5/Scratch/Anton/Testing/15825
  - Plot the monitors (1-10) and confirm that there is chopped up data for some of them (spectrum 4). This is best seen by setting the axes to log-log.
- Apply the `UnwrapMonitorsInTOF` algorithm with only the workspace set
  - Confirm that the output workspace has now twice the TOF range and the data is appended to itself
- Apply the `UnwrapMonitorsInTOF` algorithm with setting the range to 2 to 15
  - Confirm that for monitor 4 only the central (physical sensible) bit is left. This is best seen by setting the axes to log-log.

**Release Notes**

See [here](https://github.com/mantidproject/mantid/pull/17371/commits/a963c8ed0dab75d738dd84ef2ad12486a73b3bfa)

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
